util/cache.go

```diff
@@ -0,0 +1,111 @@
+// Handle the password caching
+package util
+
+import (
+	"errors"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+type CachedUser struct {
+	Password   string
+	UsageCount int
+	LoggedIn   bool
+}
+
+type VaultCache struct {
+	store *cache.Cache
+}
+
+var UserVaultCache *VaultCache
+
+func NewVaultCache() *VaultCache {
+	var vc *VaultCache
```

Comment:
you don't need to declare this var, just do the assignment with := on the next line